Irina_Loginova_HomeWork_2_1

Loginova-irina opened the pull request.

Loginova-irina committed:
Сумма и ср. арифметическае числа

MykolaSuperman commented on Irina_Loginova_HomeWork_2_1.cpp:
```diff
@@ -0,0 +1,24 @@
+    x = a % 10;
+  }
+  std::cout << "The sum of numbers is " << sum << std::endl;
+  std::cout << "The average is " << sum / i << std::endl;
```
тут вы не получите вещественную часть, т.к. у вас тут целочисленное деление, у вас надо чтобы один из делимых был преобразован к вещественному типу

Loginova-irina replied:
Так это не логично. Мне нужен остаток от деления двух целых чисел (a%10), Остаток тоже будет целое число, соответственно. В задаче не было сказано, что надо посчитать сумму всех чисел ВЕЩЕСТВЕННОГО числа: т.е. и чисел дробной и целой части.  Тем более предложенный пример как раз показывает, что нам надо работать с целыми числами

Loginova-irina replied:
Да и Visual Studio выдает ошибку, если какую то из переменных сделать вещественным числом, а не целым.

MykolaSuperman replied:
Всмысле не логично?)
У вас задача посчитать сумму цифр и среднее арифметическое.
Т.е. если у вас есть число `23`
```
сумма 5
потом среднее арифметическое 
5 / 2 = 2.5

в вашем случае это будет 2, т.е. вы потеряли вещественную часть
```
любое среднеарифметическое подразумевает дробную часть, т.е. вещественное выражение.
Поэтому в задаче и не говорилось о вещественном типе, т.е. среднеарифметичекое и так понимает по дефолту вещественную логику.

Loginova-irina replied:
Да. Это я поумничала про выражение: a % 10. Компилятор говорит, что а должно быть int. Я поняла про что Вы. Сейчас перепишу.

MykolaSuperman commented on Irina_Loginova_HomeWork_2_1.cpp:
```diff
@@ -0,0 +1,24 @@
+  while (x != 0) {
+    std::cout << "x= " << x << std::endl;
+    sum += x;
+    i = i++;
```
а зачем вам счетчик? у вас условие итерации цикла, есть то пока у вас x != 0. Счетчик применим тогда когда вам нужно выполнить некоторое заранее известное количество операций, и счетчик служит концом итераций

Loginova-irina replied:
Счетчик - это Вы про i? Так он мне нужен, что бы посчитать среднеарифметическое. Откуда я знаю какой длины пользователь захочет ввести число. Вот переменная i считает мне их колличество

MykolaSuperman replied:
окей, согласен. Но я бы написал бы цикл for, был бы более читабельный.
потом выражение
```
i = i++;
```
я немного не понял, зачем вам сохранять предыдущее значение?
почему не i++? или ++i

Loginova-irina replied:
i = i++ это моя невнимательность. Путаюсь я еще с --крементами. А цикл for надо писать, когда знаешь От и ДО (если я правильно поняла). А из за того, что пользователь непредсказуем, у меня ДО не известно и я написала цикл while.

MykolaSuperman replied:
Хорошо

MykolaSuperman commented on Irina_Loginova_HomeWork_2_1.cpp:
```diff
@@ -0,0 +1,24 @@
+  std::cout << "Inpot A= " << std::endl;
+  long long a{};
+  std::cin >> a;
+  int x = a % 10;
```
я думаю что эту операцию вам делать тут не надо. это должно проходить  в цикле

Loginova-irina replied:
Почему? Мне же надо циклу дать значение для сравнения - while (x != 0) . Что я ему дам, если у меня еще нет в наличие значения переменой x? Нулевое значение дать не могу, цикл сразу закроется. Поэтому я первое значение для х получила до цикла.

MykolaSuperman replied:
ну я о том что можно напиасать так
```
  while (x = a % 10) {
    std::cout << "x= " << x << std::endl;
    sum += x;
    i = i++;
....
}
```

Loginova-irina replied:
Прикольно. Работает :-). Сейчас выложу откорректированную версию.

Loginova-irina committed:
Update Irina_Loginova_HomeWork_2_1.cpp

Обновила. Убрала лишнее.

Loginova-irina committed:
Update Irina_Loginova_HomeWork_2_1.cpp

Внесла изменения в цикл. Убрала вывод цифр числа, по условию это не надо делать.
